Lib/typing.py

```diff
@@ -1696,11 +1749,12 @@ def _get_protocol_attrs(cls):
             continue
         annotations = getattr(base, '__annotations__', {})
         for attr in (*base.__dict__, *annotations):
-            if not attr.startswith('_abc_') and attr not in EXCLUDED_ATTRIBUTES:
+            if attr not in EXCLUDED_MEMBERS and not attr.startswith('_abc_'):
```

Comment:
Reversing the order seemed a teeny bit faster, since I'd assume the condition is more common to fail due to being an excluded attribute.

Upon measuring, not sure if it is not just on the noise level

```python
_TYPING_INTERNALS = frozenset({
    '__parameters__', '__orig_bases__',  '__orig_class__',
    '_is_protocol', '_is_runtime_protocol', '__protocol_attrs__',
    '__callable_proto_members_only__', '__type_params__',
})

_SPECIAL_NAMES = frozenset({
    '__abstractmethods__', '__annotations__', '__dict__', '__doc__',
    '__module__', '__slots__', '__match_args__', '__qualname__',
})
_SPECIAL_CALLABLE_NAMES = frozenset({
    '__init__', '__new__', '__subclasshook__','__class_getitem__', '__weakref__',
})

# These special attributes will be not collected as protocol members.
EXCLUDED_ATTRIBUTES = _TYPING_INTERNALS | _SPECIAL_NAMES | {'_MutableMapping__marker'}
EXCLUDED_MEMBERS = EXCLUDED_ATTRIBUTES | _SPECIAL_CALLABLE_NAMES

def get_members(base):
    attrs = set()
    annotations = getattr(base, '__annotations__', {})
    for attr in (*base.__dict__, *annotations):
        if not attr.startswith('_abc_') and attr not in EXCLUDED_MEMBERS:
            attrs.add(attr)
    return attrs

def get_members2(base):
    attrs = set()
    annotations = getattr(base, '__annotations__', {})
    for attr in (*base.__dict__, *annotations):
        if attr not in EXCLUDED_MEMBERS and not attr.startswith('_abc_'):
            attrs.add(attr)
    return attrs

%timeit get_members(tuple)
%timeit get_members2(tuple)
%timeit get_members(list)
%timeit get_members2(list)
%timeit get_members(dict)
%timeit get_members2(dict)
from pandas import Series, DataFrame
%timeit get_members(Series)
%timeit get_members2(Series)
%timeit get_members(DataFrame)
%timeit get_members2(DataFrame)
```

Reply:
3.09 µs ± 23.2 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
2.99 µs ± 40.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
4.84 µs ± 90.6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
4.64 µs ± 244 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
6.39 µs ± 421 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
5.76 µs ± 229 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
28.9 µs ± 626 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
28 µs ± 450 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
32.4 µs ± 752 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
31.9 µs ± 420 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

Reply:
This function is only ever called at class creation time. Can you show me a benchmark that shows that changing the order here makes a measurable difference to the time it takes to create a Protocol subclass? Unless the performance speedup is visible to users, I don't think it's worth making the change here.

Even if it is a measurable speedup, I think it should probably go in its own PR, to keep the diff for this PR clean